crates/miden-lib/src/account/interface/mod.rs

```diff
@@ -257,9 +261,103 @@ impl AccountComponentInterface {
+                    push.{aux}
+                    push.{tag}
+                    ",
+                recipient = word_to_felts_string(&partial_note.recipient_digest()),
+                note_type = Felt::from(partial_note.metadata().note_type()),
+                execution_hint = Felt::from(partial_note.metadata().execution_hint()),
+                aux = partial_note.metadata().aux(),
+                tag = Felt::from(partial_note.metadata().tag()),
+            ));
+
+            match self {
+                AccountComponentInterface::BasicFungibleFaucet => {
+                    if partial_note.assets().num_assets() != 1 {
+                        return Err(TransactionScriptBuilderError::FaucetNoteWithoutAsset);
+                    }
+
+                    // SAFETY: We checked that the note contains exactly one asset
+                    let asset =
+                        partial_note.assets().iter().next().expect("note should contain an asset");
+
+                    if asset.faucet_id_prefix() != sender_account_id.prefix() {
+                        return Err(TransactionScriptBuilderError::IssuanceFaucetMismatch(
+                            asset.faucet_id_prefix(),
+                        ));
+                    }
+
+                    body.push_str(&format!(
+                        "
+                        push.{amount}
+                        call.::miden::contracts::faucets::basic_fungible::distribute dropw dropw drop
+                        ",
+                        amount = asset.unwrap_fungible().amount()
+                    ));
+                },
+                AccountComponentInterface::BasicWallet => {
+                    body.push_str(
+                        "
+                        call.::miden::contracts::wallets::basic::create_note\n",
+                    );
+
+                    for asset in partial_note.assets().iter() {
+                        body.push_str(&format!(
+                            "push.{asset}
+                            call.::miden::contracts::wallets::basic::move_asset_to_note dropw\n",
+                            asset = word_to_felts_string(&asset.into())
+                        ));
+                    }
+
+                    body.push_str("dropw dropw dropw drop");
+                },
+                _ => return Err(TransactionScriptBuilderError::UnsupportedInterface(self.clone())),
+            }
+        }
+
+        Ok(body)
+    }
 }
 
-// ACCOUNT COMPONENT INTERFACE
+impl Display for AccountComponentInterface {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        match self {
+            AccountComponentInterface::BasicWallet => write!(f, "Basic Wallet"),
+            AccountComponentInterface::BasicFungibleFaucet => write!(f, "Basic Fungible Faucet"),
+            AccountComponentInterface::RpoFalcon512(_) => write!(f, "RPO Falcon512"),
+            AccountComponentInterface::Custom(_) => write!(f, "Custom"),
```

Comment:
Should we maybe include the procedure roots in the string representation? E.g. a custom component with two procedures would result in a string like `Custom([0x1ce254c1..., 0x8821e7e1...])`. Taking the first 10 hex characters or so should be sufficient for debugging purposes and improves readability, and we can indicate its shortened with `...`.

Reply:
I agree, good idea. But probably using 10 characters will make the resulting string quite long, I would try to use 7, as it does github for the commit hashes, probably it will be more familiar format for the user and won't take too much space.

Reply:
Yeah, shorter is also fine I guess. I was also looking at GitHub short hashes yesterday for this :smile:. My `git log --oneline` shows 8 hex characters, but anything from 7 to 10 is fine I think.

Sounds good, maybe the ellipsis is not necessary.